docs/modules/publisher-directmethods.md

```diff
@@ -0,0 +1,427 @@
+  >             "OutgressInputBufferDropped":"0",
+  >             "OutgressIoTMessageCount":"0",
+  >             "ConnectionRetries":"0",
+  >             "OpcEndpointConnected":"true",
+  >             "MonitoredOpcNodesSucceededCount":"5",
+  >             "MonitoredOpcNodesFailedCount":"0"
+  >          }
+  >       ]
+  >    }
+  >   ```
+
+## AddOrUpdateEndpoints_V1
+
+This method enables to perform a complete `published_nodes.json` update as well as update multiple
+endpoints (DataSets) at once. Unlike `PublishNodes_V1` method, `AddOrUpdateEndpoints_V1`  completely
+changes the node set for an endpoint (DataSet) with the one provided in the method's request payload.
+Furthermore, by providing an empty list of nodes in the request, the user can remove completely the
+previously configured nodes for a specific endpoint (DataSet).
+
+  _Request_: represents a list of objects which should strictly follow the request payload schema as
+  described above. The `OpcNodes` attribute being empty list or `null` will be interpreted as a removal
+  request for that endpoint (DataSet).
+
+  _Response_: when successful -  Status 200 and an empty json (`{}`) as payload
+
+  _Exceptions_: a response corresponding to an exception will be returned if:
+
+  - request payload contains deletion request for an endpoint (DataSet) that is not present in publisher configuration
+  
+  - request payload contains two or more entries for the same endpoint (DataSet)
+
+  _Example_:
+  > _Method Name_: `AddOrUpdateEndpoints_V1`
+  >
+  > _Payload_:
+  >
+  > ```json
+  > [
+  >       {
+  >          "EndpointInfo":{
+  >             "EndpointUrl":"opc.tcp://opcplc:50000/",
+  >             "DataSetWriterGroup":"Server0",
+  >             "UseSecurity":"false",
+  >             "OpcAuthenticationMode":"UsernamePassword",
+  >             "OpcAuthenticationUsername":"Usr"
+  >          },
+  >          "SentMessagesPerSec":"2.6",
+  >          "IngestionDuration":"{00:00:25.5491702}",
+  >          "IngressDataChanges":"25",
+  >          "IngressValueChanges":"103",
+  >          "IngressBatchBlockBufferSize":"0",
+  >          "EncodingBlockInputSize":"0",
+  >          "EncodingBlockOutputSize":"0",
+  >          "EncoderNotificationsProcessed":"83",
+  >          "EncoderNotificationsDropped":"0",
+  >          "EncoderIoTMessagesProcessed":"2",
+  >          "EncoderAvgNotificationsMessage":"41.5",
+  >          "EncoderAvgIoTMessageBodySize":"6128",
+  >          "EncoderAvgIoTChunkUsage":"1.158034",
+  >          "EstimatedIoTChunksPerDay":"13526.858105160689",
+  >          "OutgressBatchBlockBufferSize":"0",
+  >          "OutgressInputBufferCount":"0",
+  >          "OutgressInputBufferDropped":"0",
+  >          "OutgressIoTMessageCount":"0",
+  >          "ConnectionRetries":"0",
+  >          "OpcEndpointConnected":"true",
+  >          "MonitoredOpcNodesSucceededCount":"5",
+  >          "MonitoredOpcNodesFailedCount":"0"
+  >       }
+  >    ]
+  >    }
```

Comment:
Extra closing bracket. Should be removed.

Reply:
fixed